biblib/views/user_view.py

```diff
@@ -152,17 +147,13 @@ def get_libraries(cls, service_uid, absolute_uid, start=0, rows=None, sort_col="
                     num_users=num_users,
                     owner=owner
                 )
+                current_app.logger.debug('Type: {0}.'.format(type))
+                if type == 'all' or (type == 'owner' and main_permission in ['owner']) or (type == 'collaborator' and main_permission in ['admin', 'read', 'write']): 
+                    libraries.append(payload)
 
-                if (ownership and main_permission in ['owner']) or not ownership: 
-                    my_libraries.append(payload)
-                elif ownership and main_permission in ['admin', 'read', 'write']: 
-                    shared_with_me.append(payload)
-            
-            if ownership: 
-                libraries_response['my_libraries'] = my_libraries 
-                libraries_response['shared_with_me'] = shared_with_me
-            else: 
-                libraries_response['libraries'] = my_libraries
+            if rows: libraries = libraries[start: start+rows]
+            elif start > 0: libraries = libraries[start:]
+            libraries_response = {'count': len(libraries), 'libraries': libraries}
```

Comment:
Here the count is not the count of all (all | shared | owned) libraries but only number of libraries returned in the response, right? Shouldn't it be counted before line 154?

Reply:
Ops, I was only looking at the first commit. I see you fixed that in second commit.